supertokens_python/recipe/thirdpartypasswordless/recipe.py

```diff
@@ -55,36 +65,43 @@
     CreateEmailVerificationTokenEmailAlreadyVerifiedError,
     CreateEmailVerificationTokenOkResult)
 from ..emailverification.interfaces import RecipeInterface as EVRecipeInterface
+from ..emailverification.interfaces import \
+    TypeEmailVerificationEmailDeliveryInput
 from ..emailverification.utils import OverrideConfig as EVOverrideConfig
 from ..passwordless import PasswordlessRecipe
 from ..passwordless.interfaces import APIInterface as PasswordlessAPIInterface
 from ..passwordless.interfaces import \
     RecipeInterface as PasswordlessRecipeInterface
+from ..passwordless.interfaces import TypePasswordlessEmailDeliveryInput
 from ..passwordless.utils import OverrideConfig as PlessOverrideConfig
 from ..thirdparty.interfaces import APIInterface as ThirdPartyAPIInterface
 from ..thirdparty.interfaces import \
     RecipeInterface as ThirdPartyRecipeInterface
 from .exceptions import SupertokensThirdPartyPasswordlessError
 from .interfaces import APIInterface, RecipeInterface
+from .types import TypeThirdPartyPasswordlessEmailDeliveryInput
 from .utils import InputOverrideConfig, validate_and_normalise_user_input
 
 
 class ThirdPartyPasswordlessRecipe(RecipeModule):
     recipe_id = 'thirdpartypasswordless'
     __instance = None
+    email_delivery: EmailDeliveryIngredient[TypeThirdPartyPasswordlessEmailDeliveryInput]
 
     def __init__(self, recipe_id: str, app_info: AppInfo,
                  contact_config: ContactConfig,
                  flow_type: Literal['USER_INPUT_CODE', 'MAGIC_LINK', 'USER_INPUT_CODE_AND_MAGIC_LINK'],
-                 get_link_domain_and_path: Union[Callable[[
-                     PhoneOrEmailInput, Dict[str, Any]], Awaitable[str]], None] = None,
+                 ingredients: ThirdPartyPasswordlessIngredients,
```

Comment:
Why are you taking ingredients in the middle here but in other recipes, it's at the end of the params list?

Reply:
Python doesn't allow you to pass optional args before mandatory ones. `ingredients` is mandatory (cannot be None). So it must appear before the optional args. 

> but in other recipes, it's at the end of the params list?

Previously it was optional. Now, it isn't. So it must appear before the optional args.